runtime/near-evm-runner/src/lib.rs

```diff
@@ -398,6 +483,68 @@ impl<'a> EvmContext<'a> {
             .ok_or(VMLogicError::EvmError(EvmError::IntegerOverflow))?;
         self.gas_counter.pay_action_accumulated(burn_gas, use_gas, ActionCosts::new_receipt)
     }
+
+    fn pay_gas_from_evm_gas(&mut self, op: EvmOpForGas) -> Result<()> {
+        let fee_cfg = &self.fees_config.evm_config;
+        let evm_gas = self.evm_gas_counter.used_gas.as_u64();
+        self.gas_counter.inc_evm_gas_counter(evm_gas);
+        let gas = match op {
+            EvmOpForGas::Deploy => {
+                evm_gas * fee_cfg.deploy_cost_per_evm_gas + fee_cfg.bootstrap_cost
+            }
+            EvmOpForGas::Funcall => {
+                evm_gas * fee_cfg.funcall_cost_per_evm_gas
+                    + fee_cfg.funcall_cost_base
+                    + fee_cfg.bootstrap_cost
+            }
+            EvmOpForGas::Other => fee_cfg.bootstrap_cost,
+        };
+        self.gas_counter.pay_evm_gas(gas)
+    }
+}
+
+fn max_evm_gas_from_near_gas(
+    near_gas: Gas,
+    evm_gas_config: &EvmCostConfig,
+    method_name: &str,
+) -> Option<U256> {
+    match method_name {
+        "deploy_code" => {
+            if near_gas < evm_gas_config.bootstrap_cost {
+                return None;
+            }
+            Some(
+                ((near_gas - evm_gas_config.bootstrap_cost)
+                    / evm_gas_config.deploy_cost_per_evm_gas)
+                    .into(),
+            )
+        }
+        "call_function" => {
+            if near_gas < evm_gas_config.bootstrap_cost + evm_gas_config.funcall_cost_base {
+                return None;
+            }
+            Some(
+                ((near_gas - evm_gas_config.bootstrap_cost - evm_gas_config.funcall_cost_base)
+                    / evm_gas_config.funcall_cost_per_evm_gas)
+                    .into(),
+            )
+        }
+        _ => {
+            if near_gas < evm_gas_config.bootstrap_cost {
+                return None;
+            }
+            Some(evm_gas_config.bootstrap_cost.into())
+        }
+    }
+}
+
+#[cfg(feature = "costs_counting")]
+pub fn evm_last_deployed_addr() -> H160 {
```

Comment:
let's remove this and the constant.
the right way to do this - is to query result of the transaction.
we are doing this across all tests - like this https://github.com/nearprotocol/nearcore/blob/965265edf114cea435d03304dcd901a46f149478/test-utils/testlib/src/standard_test_cases.rs#L1290

Reply:
OK

Reply:
Seems i cannot use testlib::RuntimeNode in an obvious way, there is only runtime in testbed, so far didn't successfully construct one from runtime, will continue tomorrow

Reply:
Done